### PR TITLE
do not raise Error but set VariantType to Null if value is None in Va…

### PR DESCRIPTION
**Pull request by oroulet**
…riant constrcture

### CODE REVIEW

File: asyncua/ua/uatypes.py [hunk 1]
```diff
@@ -750,7 +750,7 @@ def __init__(self, value=None, varianttype=None, dimensions=None, is_array=None)
         if self.VariantType is None:
             self.VariantType = self._guess_type(self.Value)
         if self.Value is None and not self.is_array and self.VariantType not in (VariantType.Null, VariantType.String, VariantType.DateTime, VariantType.ExtensionObject):
-            raise UaError(f"Non array Variant of type {self.VariantType} cannot have value None")
+            self.VariantType == VariantType.Null
```
Comment (AndreasHeine) [inline asyncua/ua/uatypes.py:753]: @oroulet sorry this breaks stuff! get a lot off errors! if i browse some nodes...

Comment (AndreasHeine) [reply asyncua/ua/uatypes.py:753]: Traceback (most recent call last):
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\server\uaprocessor.py", line 126, in process_message
    return await self._process_message(typeid, requesthdr, seqhdr, body)
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\server\uaprocessor.py", line 222, in _process_message
    self.send_response(requesthdr.RequestHandle, seqhdr, response)
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\server\uaprocessor.py", line 46, in send_response
    struct_to_binary(response), message_type=msgtype, request_id=seqhdr.RequestId)
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\ua\ua_binary.py", line 243, in struct_to_binary
    packet.append(list_to_binary(uatype[6:], val))
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\ua\ua_binary.py", line 283, in list_to_binary
    pack = [to_binary(uatype, el) for el in val]
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\ua\ua_binary.py", line 283, in <listcomp>
    pack = [to_binary(uatype, el) for el in val]
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\ua\ua_binary.py", line 261, in to_binary
    return pack_uatype(vtype, val)
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\ua\ua_binary.py", line 181, in pack_uatype
    return struct_to_binary(value)
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\ua\ua_binary.py", line 248, in struct_to_binary
    packet.append(to_binary(uatype, val))
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\ua\ua_binary.py", line 261, in to_binary
    return pack_uatype(vtype, val)
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\ua\ua_binary.py", line 179, in pack_uatype
    return variant_to_binary(value)
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\ua\ua_binary.py", line 364, in variant_to_binary
    b.append(pack_uatype(var.VariantType, var.Value))
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\ua\ua_binary.py", line 177, in pack_uatype
    return nodeid_to_binary(value)
  File "C:\Users\HeineAn\AppData\Local\Programs\Python\Python37-32\lib\site-packages\asyncua\ua\ua_binary.py", line 289, in nodeid_to_binary
    if nodeid.NodeIdType == ua.NodeIdType.TwoByte:
AttributeError: 'NoneType' object has no attribute 'NodeIdType'

Comment (oroulet) [reply asyncua/ua/uatypes.py:753]: how do you get these error? I tried against some different servers and clients and I cannot get any error....

Comment (oroulet) [reply asyncua/ua/uatypes.py:753]: also the error above requires a nodeid that is None. I cannot see how my patch could create such a thing...

Comment (oroulet) [reply asyncua/ua/uatypes.py:753]: Maybe it is your own code that create some NodeIds with Null value?

Comment (AndreasHeine) [reply asyncua/ua/uatypes.py:753]: browse -> BaseObjectType -> TopologyElementType -> ParameterSet -> ParameterIdentifier
its part of Opc.Ua.Di.NodeSet2.xml

code below!

**Comment by AndreasHeine** [comment]
[Server.zip](https://github.com/FreeOpcUa/opcua-asyncio/files/5480446/Server.zip)

File: asyncua/ua/uatypes.py (hunk 1 above)
Comment (oroulet) [reply asyncua/ua/uatypes.py:753]: So you only get the error when browsing that specific node?

Comment (AndreasHeine) [reply asyncua/ua/uatypes.py:753]: also with some other of that kind of placeholder variables in CS-Nodesets!

Unbenannt

**Pull request closed without merging**